### PR TITLE
nuclear: init at 0.6.6

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Building from source is a pain due to electron, so for now I've used the provided appimage.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/nuclear/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "nuclear";
+  version = "0.6.6";
+  name = "${pname}-v${version}";
+
+  src = fetchurl {
+    url = "https://github.com/nukeop/nuclear/releases/download/v${version}/${name}.AppImage";
+    sha256 = "0c1335m76fv0wfbk07s8r6ln7zbmlqd66052gqfisakl8a1aafl6";
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit name src; };
```
Comment (sikmir) [inline pkgs/applications/audio/nuclear/default.nix:12]:
```suggestion
  appimageContents = appimageTools.extract { inherit name src; };
```
See https://github.com/NixOS/nixpkgs/blob/a6b361d6276b7c413ffea0a28a1a3259d2275e2e/pkgs/build-support/appimage/default.nix#L35-L37

File: pkgs/applications/audio/nuclear/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+in appimageTools.wrapType2 {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+
+    install -m 444 -D ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
```
Comment (sikmir) [inline pkgs/applications/audio/nuclear/default.nix:19]:
```suggestion
    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
```

File: pkgs/applications/audio/nuclear/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=nuclear'
```
Comment (sikmir) [inline pkgs/applications/audio/nuclear/default.nix:21]:
```suggestion
      --replace "Exec=AppRun" "Exec=$out/bin/nuclear"
```

File: pkgs/applications/audio/nuclear/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ appimageTools, lib, fetchurl }:
+let
+  pname = "nuclear";
+  version = "0.6.6";
+  name = "${pname}-v${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/nuclear/default.nix:5]: name is auto generated and should not contain the v before the version.

File: pkgs/applications/audio/nuclear/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=nuclear'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  meta = with lib; {
+    description = "Streaming music player that finds free music for you";
+    homepage = "https://nuclear.js.org/";
+    license = licenses.agpl3Plus;
+    maintainers = [ maintainers.ivar ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/nuclear/default.nix:29]:
```suggestion
    maintainers = with maintainers; [ ivar ];
```

File: pkgs/applications/audio/nuclear/default.nix (hunk 4 above)
Comment (sikmir) [reply pkgs/applications/audio/nuclear/default.nix:5]: @SuperSandro2000 `appimageTools.*` doesn't auto generate `name`, that would be true for `mkDerivation`.

File: pkgs/applications/audio/nuclear/default.nix (hunk 5 above)
Comment (IvarWithoutBones) [reply pkgs/applications/audio/nuclear/default.nix:29]: Why? The full maintainer list doesn't need to be in scope, and this looks a lot nicer IMO.

Comment (SuperSandro2000) [reply pkgs/applications/audio/nuclear/default.nix:29]: It makes adding new maintainers easier.

Comment (IvarWithoutBones) [reply pkgs/applications/audio/nuclear/default.nix:29]: Well yes, but a second maintainer can also make that change when it is actually useful.

Comment (SuperSandro2000) [reply pkgs/applications/audio/nuclear/default.nix:29]: You can also be nice and do it in the beginning. 🙃

Comment (IvarWithoutBones) [reply pkgs/applications/audio/nuclear/default.nix:29]: In my opinion this looks way nicer than your proposed solution, which as of now has no functional advantage. 

There is no guideline on this (as far as I'm aware), so I think this should be fine.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nuclear</li>
  </ul>
</details>

**Pull request merged**